cloudconfig.go

```diff
@@ -8,24 +8,36 @@ import (
 
 	"github.com/giantswarm/clustertpr"
 	"github.com/giantswarm/clustertpr/node"
+	"k8s.io/client-go/kubernetes"
 )
 
+// Params are the parameters which can be defined by the concrete operator which is using that library.
 type Params struct {
 	Cluster   clustertpr.Cluster
 	Extension Extension
 	Node      node.Node
 }
 
+// internalParams contain both parameters defined by the operator and those defined and used only in
+// this library.
+type internalParams struct {
+	Token string
+	Params
+}
+
 type CloudConfig struct {
 	config   string
-	params   Params
+	params   internalParams
 	template string
 }
 
-func NewCloudConfig(template string, params Params) (*CloudConfig, error) {
+func NewCloudConfig(k8sClient kubernetes.Interface, template string, params Params) (*CloudConfig, error) {
```

Comment:
The PR description stated that the cloud config stuff moved to ignition. Why do we need this here then still? Also, the `k8sClient` is not used in here.